src/GraphQL/Types/Fields/FieldType.cs

```diff
@@ -58,10 +58,22 @@ public object DefaultValue
             }
         }
 
+        private Type _type;
         /// <summary>
         /// Gets or sets the graph type of this field.
         /// </summary>
-        public Type Type { get; set; }
+        public Type Type
+        {
+            get => _type;
+            set
+            {
+                if (value != null && !value.IsGraphType())
+                    throw Create(value);
+                _type = value;
+            }
+        }
+
+        private ArgumentOutOfRangeException Create(Type type) => new ArgumentOutOfRangeException("value", $"Type '{type}' is not a graph type");
```

Comment:
I assume this pattern (that I've seen elsewhere) is for performance??? Hope I did that right.

Reply:
https://reubenbond.github.io/posts/dotnet-perf-tuning

Almost.

Reply:
I don't have time to read up on this today.  Suggest a change if you think it should change.

Reply:
`throw` needs to be inside helper method so JIT can inline the rest.

Reply:
Makes sense.  Why is it this way elsewhere then?

Reply:
Is this wrong:
https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Types/QueryArgument.cs#L104..L127

Reply:
Looks like CheckType couldn't be inlined, so why it's in a private method I don't know.

Reply:
This is a low-level optimization and only makes sense on a hot path. There is absolutely no need to apply this approach everywhere. I want to say that it is quite possible to do without it here. I used throw helpers in the parser because it showed the real effect there. Not very big.